test/boost/compaction_group_test.cc

```diff
@@ -0,0 +1,147 @@
+    virtual sstables::sstables_manager& get_sstables_manager() noexcept override { return _sst_man; }
+    virtual sstables::shared_sstable make_sstable() const override { return _sstable_factory(); }
+    virtual sstables::sstable_writer_config configure_writer(sstring origin) const override { return _sst_man.configure_writer(std::move(origin)); }
+    virtual api::timestamp_type min_memtable_timestamp() const override { return api::min_timestamp; }
+    virtual future<> on_compaction_completion(sstables::compaction_completion_desc desc, sstables::offstrategy offstrategy) override {
+        testlog.info("Adding {} sstable(s), removing {} sstables", desc.new_sstables.size(), desc.old_sstables.size());
+        rebuild_main_set(desc.new_sstables, desc.old_sstables);
+        return make_ready_future<>();
+    }
+    virtual bool is_auto_compaction_disabled_by_user() const noexcept override { return false; }
+    virtual bool tombstone_gc_enabled() const noexcept override { return false; }
+    virtual const tombstone_gc_state& get_tombstone_gc_state() const noexcept override { return _tombstone_gc_state; }
+    virtual compaction_backlog_tracker& get_backlog_tracker() override { return _backlog_tracker; }
+    virtual const std::string get_group_id() const noexcept override { return "0"; }
+    virtual seastar::condition_variable& get_staging_done_condition() noexcept override { return _staging_done_condition; }
+};
+
+SEASTAR_TEST_CASE(basic_compaction_group_splitting_test) {
+    return test_env::do_with_async([] (test_env& env) {
+        auto builder = schema_builder("tests", "compaction_group_splitting")
+                .with_column("id", utf8_type, column_kind::partition_key)
+                .with_column("cl", int32_type, column_kind::clustering_key)
+                .with_column("value", int32_type);
+        auto s = builder.build();
+
+        auto t = env.make_table_for_tests(s);
+        auto close_table = deferred_stop(t);
+        t->start();
+
+        auto sst_factory = env.make_sst_factory(s);
+
+        auto compaction_group = std::make_unique<single_compaction_group>(t, env.manager(), sst_factory);
+
+        auto input = sstable_that_needs_split(s, sst_factory);
+
+        compaction_group->rebuild_main_set({std::move(input)}, {});
+
+        auto classifier = [] (dht::token t) -> mutation_writer::token_group_id {
+            return dht::compaction_group_of(1, t);
+        };
+
+        auto& cm = t->get_compaction_manager();
+
+        cm.perform_splitting_compaction(*compaction_group, sstables::compaction_type_options::splitting{classifier}).get();
+
+        BOOST_REQUIRE(compaction_group->main_sstable_set().size() == 2);
+        compaction_group->main_sstable_set().for_each_sstable([&] (const sstables::shared_sstable& sst) {
+            BOOST_REQUIRE_EQUAL(classifier(sst->get_first_decorated_key().token()), classifier(sst->get_last_decorated_key().token()));
+        });
```

Comment:
Should also require that the classifier returns the correct value, not just that they are equal.

Also, for the splitter test: let's test that if all the input tokens are in the left side, or all in the right side, that we end up with one sstable in the correct side.